crates/uv-python/src/interpreter.rs

```diff
@@ -138,6 +139,50 @@ impl Interpreter {
         Ok(base_python)
     }
 
+    /// Attempt to derive a path from base Python that substitutes a minor
+    /// version symlink directory (or junction on Windows) for the patch version
+    /// directory.
+    ///
+    /// If this interpreter is PyPy or GraalPy, return [`None`].
+    pub fn maybe_symlink_path_from_base_python(&self, base_python: &Path) -> Option<PathBuf> {
+        if self.markers().implementation_name() == "pypy"
```

Comment:
This is accessible as `self.implementation_name()` too, but if we're doing comparisons we should probably use the typed version, like

https://github.com/astral-sh/uv/blob/0593b967ba848909c46b4c95f4ba8cbdcba147e9/crates/uv-python/src/installation.rs#L215-L218

(we'd need to move that method into `Interpreter`)